packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -207,25 +196,60 @@ export abstract class InputMediaDeviceManager<
     }
   }
 
-  protected async unmuteStream() {
+  protected async muteOrDestroyStream(stopTracks: boolean = true) {
+    if (!this.state.mediaStream) return;
+    this.logger('debug', `${stopTracks ? 'Stopping' : 'Disabling'} stream`);
+    if (this.call.state.callingState === CallingState.JOINED) {
+      await this.stopPublishStream(stopTracks);
+    }
+    this.muteLocalStream(stopTracks);
+    const allEnded = this.getTracks().every((t) => t.readyState === 'ended');
+    if (allEnded) {
+      if (
+        this.state.mediaStream &&
+        // @ts-expect-error release() is present in react-native-webrtc
+        typeof this.state.mediaStream.release === 'function'
+      ) {
+        // @ts-expect-error called to dispose the stream in RN
+        this.state.mediaStream.release();
+      }
+      this.state.setMediaStream(undefined);
+    }
+  }
+
+  protected async unmuteOrCreateStream(
+    settings?: UnmuteOrCreateStreamSettings,
+  ) {
     this.logger('debug', 'Starting stream');
     let stream: MediaStream;
-    if (
-      this.state.mediaStream &&
-      this.getTracks().every((t) => t.readyState === 'live')
-    ) {
-      stream = this.state.mediaStream;
-      this.unmuteTracks();
-    } else {
+    if (!this.state.mediaStream && settings && settings.createMutedStream) {
       const defaultConstraints = this.state.defaultConstraints;
       const constraints: MediaTrackConstraints = {
         ...defaultConstraints,
         deviceId: this.state.selectedDevice,
       };
       stream = await this.getStream(constraints as C);
+      this.state.setMediaStream(stream);
+      await this.muteOrDestroyStream(this.state.disableMode === 'stop-tracks');
```

Comment:
```suggestion
      await this.muteOrDestroyStream(false);
```

We always want to just mute the stream here, if we call it with `stop-tracks` we destroy the stream that we've just created